### ISSUE TITLE
Portrait videos converted to landscape on load

### ISSUE THREAD

**Issue by angyangie** [Zulko/moviepy#586]
When making a VideoFileClip, as such `video = editor.VideoFileClip(video_file)`, with a `video_file` that's portrait sized (`video_file` size is [1080, 1920]), Moviepy converts to landscape, `video` = [1920, 1080]), `video.rotation` = 90. 

I found this workaround (below), but seems like that's something that shouldn't happen.

```python 
if video.rotation == 90:
    video = video.resize(video.size[::-1])
    video.rotation = 0

**Comment by Gloin1313** [comment]
Thanks for this! The `video.rotation` is a new feature, recently implemented, and I have thought it would be nice to have another parameter for `VideoFileClip` to do with rotation, for instance, a `default_rotation` parameter, which if True would rotate the video depending on the value of `video.rotation`.

For now, though, this is working as expected.

**Comment by angyangie** [comment]
A `default_rotation` as described would definitely be useful! Thanks for the response @Gloin1313

**Comment by Gloin1313** [comment]
I'll look at getting it in soon! I'll leave this open until then.

**Comment by angyangie** [comment]
Awesome! Thank you @Gloin1313. To clarify, the problem is that `video = editor.VideoFileClip(video_file)` returns a portrait input as a landscape, automatically (i.e. the portrait video is stretched out to fit into a landscape frame, which distorts everything, though the video is still the right side up). 

To your point above, replacing `video.resize(...)` with `video.rotate(90)` did not fix the problem for me. When I tried `video.rotate(90)` without resizing, the video simply turned on its side (i.e. I would have had to turn my computer around 90 degrees to view the video the right side up). The workaround above seemed to be the only way I could fix the 'stretching out' of the video, and still have the video display the right side up. 

Thank you so much for looking into this! We're big fans of Moviepy here at BuzzFeed!

**Comment by Gloin1313** [comment]
@angyangie Oh, I didn't realise the picture was being distorted! I hadn't looked closely at what you were doing with the `resize`, and assumed it was just a way to rotate it. I'll label this as a bug then. I haven't experienced this with my own portrait videos though.

**Comment by dpmontero** [comment]
Hi,
Bug, not auto-resize with good ratio when video metadata rotate is 90º or 270º, portrait. 

While... you can to use the param `target_resolution=[420,700]` that is work to ffmpeg direct, but this not relation aspect ratio if there rotation. See `https://zulko.github.io/moviepy/_modules/moviepy/video/io/VideoFileClip.html#VideoFileClip`

Thanks!

**Comment by cvn** [comment]
I have a PR that addresses this issue, #683. My solution is to detect if ffmpeg is autorotating the video and swap the dimensions automatically. I'd welcome any and all feedback.

**Comment by Akhilrajns** [comment]
Thanks that helped us a lot :)

**Comment by KanishMathew** [comment]
It was a good solution.It worked.

**Comment by taylorfinley** [comment]
I've been using this workaround in one of my projects, with one small modification:
```
if video.rotation in (90, 270):
    video = video.resize(video.size[::-1])
    video.rotation = 0
```
Without the modification some videos were still suffering from the stretching issue.

**Comment by fergald** [comment]
Here is a minimal video that demonstrates the problem.

https://photos.app.goo.gl/v7Y58mjg1k9RL8tq8

just download the video and then try

```py
from moviepy.editor import *
concatenate_videoclips(VideoFileClip("VID_20190803_230713.mp4")).write_videofile("bad.mp4")
```

**Comment by UtkuBulkan** [comment]
Are there any updates regarding this situation ?

**Comment by KiLLAAA** [comment]
Hello there, things can get even more complicated. With camera held badly in hand at the beginning of video causing "wrong" metadata of rotaion. This cannot be fixed by workaround above, nor clip.rotate(90) which is pawned by another bug  #1278 . This case can be fixed by setting the metadata of rotation to zero ` $ avconv -i bad_metadata.mp4 -vcodec copy -metadata:s:v:0 rotate=0  -acodec copy new_metadata_rot_0.mp4`

**Comment by tburrows13** [comment]
@KiLLAAA thanks for the comment. If the video has the wrong metadata, then there's nothing that moviepy can do about it. You'll just have to manually rotate it. #1278 has now got a fix, so should be fixed in v2.0.0.dev3.

This doesn't address the original issue (which #683 might be a fix for, but needs a bit of cleaning up if it is to be merged).

**Comment by mondeja** [comment]
There is another example file with this behaviour [here](https://github.com/earney/moviepy_media/blob/master/tests/videos/ficus_vertical.mp4?raw=true), from issue #682.

**Comment by mondeja** [comment]
This should have be fixed by #577, just reopen this issue or advice if isn't already fixed.

**Comment by simonbarker** [comment]
So I think this might be a problem again?

I have a vertical video that is read in as landscape and stretched. If I try to resize it then then output isn't playable.

```
video_clip = VideoFileClip(args.video)

print(video_clip.rotation)

print(video_clip.size)

if not os.path.exists("clips"):
  os.mkdir(f"clips")

outputVideo = video_clip.subclip(5, 10)

outputVideo.write_videofile(f"clips/" + str(0) + ".mp4", codec='libx264', audio_codec='aac', temp_audiofile='temp-audio.m4a', remove_temp=True)
```

Any help would be amazing

**Comment by simonbarker** [comment]
Ok, follow up on this.

The height and width were being read in wrong by MoviePY (printed `clip.w` and `clip.h` and they were the wrong way round) even though using `mdls` command in terminal I could see the meta was correct.

I fixed by resizing with `clip_resized = short_clip.resize((1080,1920))` whatever I was doing to resize before was messing up the clip but this time it's fine 🤷‍♂️

**Comment by unlikeany-simon** [comment]
I have experienced the same issue as @simonbarker.
Improved the workaround a bit, so it detects automatically switched width and height.

`import imageio
video = imageio.get_reader(PATH_TO_VIDEO_FILE)
if video._meta['size'] != video._meta['source_size']:
    logger.warning("Video has switched width and height. Fixing...", extra={"meta": video._meta})
    clip = clip.resize(video._meta['size'])`

**Comment by rickstar666** [comment]
i met the same problem. it happends when dealing  ios viedo , their suffix is “.MOV”
it doesn't happen when dealing MP4 file

moviepy version is 1.0.3

so i have to add some codes
```  
if newclip.rotation in (90, 270):
            newclip = newclip.resize(newclip.size[::-1])
            newclip.rotation = 0 
```

**Comment by mondeja** [comment]
This is fixed in the **master** branch, not in 1.0.3
